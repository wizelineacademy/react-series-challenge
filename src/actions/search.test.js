import configureStore from "redux-mock-store";
import searchActions from "./search";

const mockStore = configureStore();
const store = mockStore();

describe("search_actions", () => {
  beforeEach(() => {
    store.clearActions();
  });

  describe("search_input", () => {
    test("Should dispatches the correct action and payload", () => {
      store.dispatch(searchActions.creators.setSearchInput("soccer"));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe("search_favorite", () => {
    test("Should dispatches the correct action and payload", () => {
      store.dispatch(searchActions.creators.setSearchFavorite("google"));
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
