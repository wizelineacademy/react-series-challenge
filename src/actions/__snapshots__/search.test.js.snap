// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search_actions search_favorite Should dispatches the correct action and payload 1`] = `
Array [
  Object {
    "query": "google",
    "type": "SEARCH_INPUT",
  },
]
`;

exports[`search_actions search_input Should dispatches the correct action and payload 1`] = `
Array [
  Object {
    "query": "soccer",
    "type": "SEARCH_INPUT",
  },
]
`;
